packages/react-native/React/Fabric/Mounting/RCTMountingManager.mm

```diff
@@ -321,7 +321,15 @@ - (void)synchronouslyUpdateViewOnUIThread:(ReactTag)reactTag
   if (props[@"transform"] &&
       !CATransform3DEqualToTransform(
           RCTCATransform3DFromTransformMatrix(newViewProps.transform), componentView.layer.transform)) {
-    componentView.layer.transform = RCTCATransform3DFromTransformMatrix(newViewProps.transform);
+        
+      CATransform3D transform = RCTCATransform3DFromTransformMatrix(newViewProps.transform);
```

Comment:
Can you extend `RCTCATransform3DFromTransformMatrix` to take a transformOrigin prop instead, so we don't have to duplicate this everywhere?